specs/rollup-node.md

```diff
@@ -251,6 +253,46 @@ However, they can still be challenged by a fault proof until the end of the faul
 
 > **TODO** the spec doesn't encode the notion of fault proof yet, revisit this (and include links) when it does
 
+## From L2 Block to L2 Output Root
+
+After processing a block the resulting outputs will need to be synchronized with L1 for trustless execution of
+L2-to-L1 messaging, such as withdrawals. Outputs are hashed in a tree structured form for minimal proof cost to any
+piece of data. This merkle-structure is defined with [SSZ], a type system for merkleization and serialization, used in
+L1 (beacon-chain). However, we replace `sha256` with `keccak256` to save gas costs in the EVM.
+
+[SSZ]: https://github.com/ethereum/consensus-specs/blob/dev/ssz/simple-serialize.md
+
+```python
+class L2Output(Container):
+  state_root: Bytes32
+  withdrawal_storage_root: Bytes32
+  latest_block: ExecutionPayload  # includes block-hash
+  history_accumulator_root: Bytes32  # Not functional yet
+  extension: Bytes32
+```
+
+The `state_root` is the Merkle-Patricia-Trie ([MPT][g-mpt]) root of all execution-layer accounts,
+also found in `latest_block.state_root`: this field is frequently used and thus elevated closer to the L2 output root,
+reducing the merkle proof depth and thus the cost of usage.
+
+The `withdrawal_storage_root` elevates the Merkle-Patricia-Trie ([MPT][g-mpt]) root of L2 Withdrawal contract storage.
+Instead of a MPT proof to the Withdrawal contract account in the account trie,
+one can directly access the MPT storage trie root, thus reducing the verification cost of withdrawals on L1.
+
+The `latest_block` is an execution-layer block of L2, represented as the [`ExecutionPayload`][ExecutionPayload] SSZ type
+defined in L1. There may be multiple blocks per L2 output root, only the latest is presented.
```

Comment:
This is a surprising comment at this point of the conversation. Under now, we're under the impression we need to derive outputs for every block, and this is done in the flow that goes from an L1 block to the L2 block.

If that's not the case, it should be precised towards the top of the section.

Reply:
I'm wondering if this should be split up to a different document? The existing content would be the derivation spec, and this could be distinct. (In a subsequent PR though.)

Reply:
The section title "From L2 Block to L2 Output Root" is definitely confusing here. I'll move it to a separate document